src/extensions/default/SVGCodeHints/main.js

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright (c) 2015 Amin Ullah Khan. All rights reserved.
+ *  
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"), 
+ * to deal in the Software without restriction, including without limitation 
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, 
+ * and/or sell copies of the Software, and to permit persons to whom the 
+ * Software is furnished to do so, subject to the following conditions:
+ *  
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *  
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, maxerr: 50 */
+/*global define, brackets */
+
+define(function (require, exports, module) {
+    "use strict";
+    
+    // Load dependencies.
+    var AppInit             = brackets.getModule("utils/AppInit"),
+        CodeHintManager     = brackets.getModule("editor/CodeHintManager"),
+        PreferencesManager  = brackets.getModule("preferences/PreferencesManager"),
+        SVGUtils            = require("./SVGUtils"),
+        SVGTags             = require("text!SVGTags.json"),
+        SVGAttributes       = require("text!SVGAttributes.json"),
+        tagData,
+        attributeData,
+        isSVGEnabled;
+
+    // Define our own pref for hinting.
+    PreferencesManager.definePreference("codehint.SVGHints", "boolean", true);
+    
+    // Preferences to control hint.
+    function _isSVGHintsEnabled() {
+        return (PreferencesManager.get("codehint.SVGHints") !== false &&
+                PreferencesManager.get("showCodeHints") !== false);
+    }
+    
+    /**
+     * Returns a list of attributes used by a tag.
+     * 
+     * @param {string} tagName name of the SVG tag.
+     * @return {Array.<string>} list of attributes.
+     */
+    function getTagAttributes(tagName) {
+        var attrs = [], tag;
+        
+        tag = tagData.tags[tagName];
+        
+        // Get element specific attributes.
+        if (tag.attributes) {
+            attrs = attrs.concat(tag.attributes);
+        }
+        // Get group based attributes.
+        if (tag.animationAddition) {
```

Comment:
It is, but I think 9 lines vs. 38 lines is a worthy benefit.
Also, my version is easier to maintain (you don't have to change anything after adding/removing groups in the JSON file) and probably even faster.

Also, with a slightly different JSON format like `"groups": ["core", "animationAddition", ...]` (which seems even more readable to me, but that's just my humble opinion), it would be even easier.

Also, it's not that we don't have those complex expressions at all, take [this as an example](https://github.com/adobe/brackets/blob/f60b243281f0a48ab315cbd375217464321d65b1/src/document/Document.js#L644-L652).